bundles/statistics/statsgrid2016/view/IndicatorFormFlyout.js

```diff
@@ -276,17 +268,18 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.view.IndicatorFormFlyout', func
             });
         });
     },
-    validateFormValues: function (data, regionValues) {
-        if (typeof (data.name) !== 'string' || data.name.length === 0) {
+    validateIndicator: function (data) {
+        const { name } = data;
+        if (typeof name !== 'string' || name.trim().length === 0) {
+            this.errorService.warn(this.locale('errors.myIndicatorNameInput'));
             return false;
         }
-
-        if (typeof regionValues.values === 'undefined' || regionValues.values.length === 0) {
-            return false;
-        }
-
+        return true;
+    },
+    validateIndicatorData: function (regionValues) {
```

Comment:
Doesn't this mean that you need to fill in data for all regions for validation to pass?

Reply:
Or does this not include the regions that user didn't fill in?

Reply:
Didn't change data validation. Only removed length check because saveIndicatorData is called only if  if valuesForm.values.length. 

Values are modified/checked in different places. Validation shouldn't get empty values:
https://github.com/oskariorg/oskari-frontend/blob/develop/bundles/statistics/statsgrid2016/view/IndicatorFormFlyout.js#L178
https://github.com/oskariorg/oskari-frontend/blob/develop/bundles/statistics/statsgrid2016/components/IndicatorDataForm.js#L131
https://github.com/oskariorg/oskari-frontend/blob/develop/bundles/statistics/statsgrid2016/view/IndicatorFormFlyout.js#L336